src/api/app/models/project.rb

```diff
@@ -1323,43 +1323,18 @@ def reopen_release_targets
     Backend::Api::Build::Project.wipe_binaries(name)
   end
 
-  def build_succeeded?(repository)
-    states = {}
-    repository_states = {}
-
-    br = Buildresult.find_hashed(project: name, view: 'summary')
-    # no longer there?
-    return false if br.empty?
-
-    br.elements('result') do |result|
-      if result['repository'] == repository
-        repository_states[repository] ||= {}
-        result.elements('summary') do |summary|
-          summary.elements('statuscount') do |statuscount|
-            repository_states[repository][statuscount['code']] ||= 0
-            repository_states[repository][statuscount['code']] += statuscount['count'].to_i
-          end
-        end
-      else
-        result.elements('summary') do |summary|
-          summary.elements('statuscount') do |statuscount|
-            states[statuscount['code']] ||= 0
-            states[statuscount['code']] += statuscount['count'].to_i
-          end
-        end
-      end
-    end
-    if repository_states.key?(repository)
-      return false if repository_states[repository].empty? # No buildresult is bad
-      repository_states[repository].each do |state, _|
-        return false if state.in?(['broken', 'failed', 'unresolvable'])
-      end
-    else
-      return false unless states.empty? # No buildresult is bad
-      states.each do |state, _|
-        return false if state.in?(['broken', 'failed', 'unresolvable'])
+  def build_succeeded?(repo_name)
+    build_result = Buildresult.find_hashed(project: name, repository: repo_name, view: 'summary', code: [:failed, :broken, :unresolvable])
```

Comment:
Please no longer use this function - it's just a wrapper around `result_swiss_knife`. Better add an explicit function to `Backend::Api::BuildResults`

Reply:
It also converts the backend response to an `XmlHash`

Reply:
Still just a wrapper

Reply:
Which isn't a bad thing per se. We use it at many places and spares us from catching backen erorrs, converting the xml, etc. What's so bad about it?

Reply:
I won't 'request changes', but note that I believe he's stronger than you: 
https://github.com/openSUSE/open-build-service/pull/5556#issuecomment-410986185

Reply:
Ah, I though you want me to directly use the `swiss_army_knife` method.